suitesparse_bindings/sprs_suitesparse_umfpack/src/lib.rs

```diff
@@ -0,0 +1,380 @@
+//! Partial interface (3rd party, unaffiliated) to `SuiteSparse`'s UMFPACK solver package,
+//! covering essentials for solving problems of the form Ax=b for real A, and for
+//! recovering LU decomposition components of A for other uses.
+//!
+//! This wrapper currently covers the double-int (DI) and double-long (DL) variations of
+//! the underlying library, while several more variations (such as for complex data type)
+//! exist in the underlying library but do not have wrappers here.
+
+use core::ptr::{null, null_mut};
+use libc::c_void;
+use sprs::{CsMatI, PermOwnedI};
+use suitesparse_umfpack_sys::*;
+
+macro_rules! umfpack_impl {
```

Comment:
Bit of a shame all this code is in a macro. We won't get any help from clippy

Reply:
Clippy does give me commentary on the contents of the macro - might be from a recent update?